### PR TITLE
686 Dependents prefill

### PR DESCRIPTION
**Pull request by jholton**
## Description of change
Prefill 686 form from EVSS Dependents service endpoint

## Testing done
specs, manual testing

## Testing planned
testing in dev/staging

#### Unique to this PR
- [x] Prefill 686 from EVSS Dependents service endpoint

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: app/models/form_profiles/va21686c.rb [hunk 1]
```diff
@@ -1,11 +1,70 @@
 # frozen_string_literal: true
 
+module VA21686c
+  class FormFullName
+    include Virtus.model
+
+    attribute :first, String
+    attribute :middle, String
+    attribute :last, String
+    # TODO:  does service ever return suffix?
```
[the remark this answers is not in the document]
Comment (lihanli) [reply app/models/form_profiles/va21686c.rb]: these are all the docs we have https://dsva.slack.com/archives/C9S42QR99/p1533069528000103

[the remark this answers is not in the document]
Comment (jholton) [reply app/models/form_profiles/va21686c.rb]: Thanks.  I don't see any reference to suffix, so I'll just remove the comment.

File: spec/models/form_profile_spec.rb [hunk 2]
```diff
@@ -591,6 +590,11 @@ def stub_methods_for_emis_data
               end
             end
           end
+          it 'returns prefilled 21-686C' do
+            VCR.use_cassette('evss/dependents/retrieve') do
```
Comment (lihanli) [inline spec/models/form_profile_spec.rb]: can you use this user for the prefill? it has more information
https://github.com/department-of-veterans-affairs/vets-api/blob/evss-686/spec/support/vcr_cassettes/evss/dependents/retrieve.yml

Comment (jholton) [reply spec/models/form_profile_spec.rb]: Done.

Comment (lihanli) [reply spec/models/form_profile_spec.rb]: can you prefill the `dependents` array using the `children` data?

File: app/models/form_profiles/va21686c.rb [hunk 3]
```diff
@@ -1,11 +1,249 @@
...
+    attribute :date_of_marriage, ScrubbedString
+    attribute :location_of_marriage, VA21686c::FormLocation
+    attribute :spouse_full_name, VA21686c::FormFullName
+  end
+
+  class FormCurrentMarriage < FormMarriage
+    include Virtus.model
+
+    attribute :spouse_social_security_number, ScrubbedString
+    attribute :spouse_has_no_ssn, Boolean
```
Comment (jholton) [inline app/models/form_profiles/va21686c.rb]: Should we prefill this for spouse and children, or just let the end user set these values via the form?
If 'yes', we should prefill `noSsnReasonType`.

File: app/models/form_profiles/va21686c.rb [hunk 4]
```diff
@@ -1,11 +1,249 @@
...
+        spouse_address: prefill_address(spouse['address']),
+        spouse_is_veteran: spouse['veteran'],
+        live_with_spouse: spouse['sameResidency'],
+        spouse_date_of_birth: convert_date(spouse['dateOfBirth'])
+      }.compact
+    )
+  end
+
+  def detect_file_number(file_number)
+    return if file_number.nil? || file_number.match(/^\d{3}-\d{2}-\d{4}$/)
```
Comment (jholton) [inline app/models/form_profiles/va21686c.rb]: `retrieve` endpoint often returns a `ssn` for the `vaFileNumber` value.

File: app/models/form_profiles/va21686c.rb [hunk 5]
```diff
@@ -1,11 +1,249 @@
...
+        country_text: country['textCountry'],
+        city: city,
+        state: state
+      }.compact
+    )
+  end
+
+  def convert_date(date)
+    return unless date
+    Date.strptime(date.to_s, '%Q').strftime('%Y-%m-%d')
```
Comment (jholton) [inline app/models/form_profiles/va21686c.rb]: Is this an appropriate date format to return?

File: app/models/form_profiles/va21686c.rb [hunk 6]
```diff
@@ -1,11 +1,249 @@
...
+        city: address['city'],
+        state: address['state'],
+        postal_code: "#{address['zipCode']}#{"-#{address['zipLastFour']}" if address['zipLastFour'].present?}",
+        country: address.dig('country', 'dropDownCountry'),
+        country_text: address.dig('country', 'textCountry'),
+        post_office: address['postOffice'],
+        postal_type: address['postalType']
+      }.compact
+    )
+    return if address.attributes.values.uniq.all? { |x| ['DOMESTIC', ''].include? x }
```
Comment (jholton) [inline app/models/form_profiles/va21686c.rb]: `retrieve` returns blank addresses with all empty attributes except `addressLocality` will be `DOMESTIC`

File: app/models/form_profiles/va21686c.rb (hunk 5 above)
Comment (jholton) [reply app/models/form_profiles/va21686c.rb]: I checked, and the format matches other date examples in `form_profile_spec`

File: app/models/form_profiles/va21686c.rb (hunk 3 above)
Comment (lihanli) [reply app/models/form_profiles/va21686c.rb]: not sure, i would ask in the 686 channels if those values can be returned by the retrieve end point

File: spec/models/form_profile_spec.rb [hunk 7]
```diff
@@ -591,6 +710,11 @@ def stub_methods_for_emis_data
               end
             end
           end
+          it 'returns prefilled 21-686C' do
+            VCR.use_cassette('evss/dependents/retrieve_user_with_max_attributes', allow_playback_repeats: true) do
```
[the remark this answers is not in the document]
Comment (jholton) [reply spec/models/form_profile_spec.rb]: Good catch.  It's not needed now.  I just needed it while I was playing around with it in a pry session.

File: app/models/form_profiles/va21686c.rb (hunk 5 above)
Comment (lihanli) [reply app/models/form_profiles/va21686c.rb]: can you change `strftime('%Y-%m-%d')` to `to_s`?

**Pull request merged**